blockchain/cancellation.go

```diff
@@ -18,7 +18,10 @@ func SleepWithContext(ctx context.Context, delay time.Duration) error {
 
 // SlowReturn combines Sleep with returning specified bool value
 func SlowReturn(ctx context.Context, delay time.Duration, value bool) bool {
-	SleepWithContext(ctx, delay)
+	err := SleepWithContext(ctx, delay)
+	if err != nil {
+		panic(err)
```

Comment:
Refactor this function to return bool, error, and return the error instead of panicking.